www/js/metrics.js

```diff
@@ -1315,9 +1313,13 @@ angular.module('emission.main.metrics',['nvd3', 'emission.services', 'ionic-date
         return ($scope.expandedc)? "expanded-calorie-card" : "small-calorie-card";
   }
 
-  $scope.openRecentTrip = function(tripId, startTs) {
-      Timeline.updateForDay(moment(startTs).startOf('day'));
-      window.location.href = "#/root/main/diary/" + tripId;
+  $scope.openRecentTrip = function(tripId, trip_ts) {
+    $rootScope.recentTripID = tripId;
+    $rootScope.recentTripDate = moment(trip_ts * 1000);
+    $rootScope.tripTimelineUpdate = true;
+    $state.go('root.main.diary').then(function() {
```

Comment:
As I said, this will only work if the recent trips are from today

> if you test that against a user who does have data for today, and replace `currTrips[0].id` with the id of the recent trip, it should work

Reply:
My original suggestion was

> it is actually a relevant use case that there are no trips over (say) the weekend, so you may want to load the timeline (in the background) for the days that correspond to the recent trips

But looking at the code again, we can only have the timeline for one day at a time right now

So a better short-term fix might be to have the list view take in the day to load, and the invoke `$state.go` with the correct params.

The long-term fix would be to have the timeline cache the data for multiple days instead of just the current day and then load multiple timeline days in the background.

Or have the timeline work for a time range instead of on a day by day basis.